process/logsevents/serialize.go

```diff
@@ -94,17 +93,57 @@ func (logsAndEventsProcessor) SerializeTokens(tokens []*data.TokenInfo) ([]*byte
 	return buffSlice.Buffers(), nil
 }
 
+func serializeToken(tokenData *data.TokenInfo) ([]byte, []byte, error) {
+	if tokenData.TransferOwnership {
+		return serializeTokenTransferOwnership(tokenData)
+	}
+
+	meta := []byte(fmt.Sprintf(`{ "index" : { "_id" : "%s" } }%s`, tokenData.Token, "\n"))
+	serializedData, err := json.Marshal(tokenData)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return meta, serializedData, nil
+}
+
+func serializeTokenTransferOwnership(tokenData *data.TokenInfo) ([]byte, []byte, error) {
+	meta := []byte(fmt.Sprintf(`{ "update" : { "_id" : "%s", "_type" : "_doc" } }%s`, tokenData.Token, "\n"))
+	tokenDataSerialized, err := json.Marshal(tokenData)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var currentOwnerData data.OwnerData
+	if len(tokenData.OwnersHistory) > 0 {
+		currentOwnerData = *tokenData.OwnersHistory[0]
+	}
+
+	ownerDataSerialized, err := json.Marshal(&currentOwnerData)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	serializedDataStr := fmt.Sprintf(`{"script": {`+
+		`"source": "if (!ctx._source.containsKey('ownersHistory')) { ctx._source.ownersHistory = [ params.elem ]; ctx._source.currentOwner = params.owner; } else {  ctx._source.currentOwner = params.owner; ctx._source.ownersHistory.add(params.elem); }",`+
```

Comment:
`ctx._source.currentOwner = params.owner;` exists on both cases. move it outside the if